### DISCUSSION TITLE
Run GDS and Fprime simulation Sample Code

### DISCUSSION THREAD

**Discussion by satzguyz** [nasa/fprime#1367] [Q&A]
Hello,
I am able to download and compile the Fprime in ubuntu. I can generate the list of lib files in the path
fprime/lib/Linux/lib*.a

Can I try any sample program to connect with Gds running in localhost machine. Any such sample available.

Few ref codes found in fprime/Ref/*. But not sure exact way to try it out.

**Comment by Emil808** [comment]
Ref/ is a complete enough deployment to run. Read the [Ref's ReadMe](https://github.com/nasa/fprime/tree/devel/Ref) for the instructions. 

Then there is the [Math Tutorial](https://github.com/nasa/fprime/tree/devel/docs/Tutorials/MathComponent). You can just Copy/Paste the listed folders into Ref then do the same instructions to generate, build and run. 
- MathPorts
- MathReceiver
- MathSender
- MathTypes
- Top
- CMakeLists.txt

**Reply by satzguyz** [reply]
Hello @Emil808 
Thanks for pointing out. I am facing issue while following the steps given there.

I am issuing the command, 

> cd fprime/Ref
> fprime-util generate

It is showing the logs as follows, 

> 
> -- Adding Library: Drv_LinuxSpiDriver
> -- Adding Library: Drv_LinuxI2cDriver
> -- Adding Library: Drv_UartFramer
> -- Adding Library: Drv_Ip
> -- Adding Library: Drv_TcpClient
> -- Adding Library: Drv_TcpServer
> -- Adding Library: Drv_Udp
> -- Adding Library: CFDP_Checksum
> -- Adding Library: Utils
> -- Adding Library: Utils_Hash
> -- Adding Library: Utils_Types
> -- Adding Library: Ref_PingReceiver
> -- Adding Library: Ref_RecvBuffApp
> -- Adding Library: Ref_SendBuffApp
> -- Adding Library: Ref_SignalGen
> -- Adding Library: Ref_Top
> -- Adding Deployment: Ref
> -- Configuring done
> -- Generating done
> -- Build files have been written to: /home/user/fprime/Ref/build-fprime-automatic-native

When I am trying to run the command

> fprime-gds --no-app

It is showing error message as follows,

> [INFO] Autodetected artifacts root '/home/user/fprime/Ref/build-artifacts' from deployment settings.ini file.
> [ERROR] dictionary location /home/user/fprime/Ref/build-artifacts/Linux/dict does not exist
>

**Reply by LeStarch** [reply]
You generated the build cache, next you should build the software:

```
fprime-util build
```

Then run GDS (with the app):

```
fprime-gds
```

**Reply by satzguyz** [reply]
Thanks it works
